src/vconsole/vconsole-setup.c

```diff
@@ -516,19 +516,27 @@ static void setup_remaining_vcs(int src_fd, unsigned src_idx, bool utf8) {
 }
 
 static int find_source_vc(char **ret_path, unsigned *ret_idx) {
-        int r, err = 0;
+        int err = 0;
```

Comment:
hmm, if all attempts to open a VC end up win EINVAL or so, then we'll return `err` as 0 now, no? that's broken (but probably pre-existing?)

Reply:
I don't think so. The code does:

```
                fd = open_terminal(path, O_RDWR|O_CLOEXEC|O_NOCTTY);
                if (fd < 0) {
                        log_debug_errno(fd, "Failed to open terminal %s, ignoring: %m", path);
                        if (IN_SET(err, 0, -EBUSY, -ENOENT))
                                err = fd;
                        continue;
                }
```
if `fd = -EINVAL` and `err = 0` then `err` is set to `-EINVAL`.

Or am I missing your point ?

Reply:
sorry, misread the code.